aperturedb/Query.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List
+
+from aperturedb.Constraints import Constraints
+from aperturedb.Sort import Sort
+
+
+class EntityType(Enum):
+    """
+    Based on : https://docs.aperturedata.io/API-Description.html
+
+    """
+    CUSTOM = ""
+    IMAGE = "_Image"
+    POLYGON = "_Polygon"
+    BOUNDING_BOX = "_BoundingBox"
+    VIDEO = "_Video"
+    BLOB = "_Blob"
+    DESCRIPTOR = "_Descriptor"
+    DESCRIPTORSET = "_DescriptorSet"
+
+
+class Query():
+    """
+    This is the underlying class to generate a query using python code.
+    """
+    db_object = "Entity"
+    find_command = f"Find{db_object}"
+    update_command = f"Update{db_object}"
+    next = None
+
+    def connected_to(self,
+                     spec: Query,
+                     adj_to: int = 0) -> Query:
+        """
+        Can be used to connect two commands together, for non trivial queries.
+        This is the lower level function to create a query.
+
+        Args:
+            spec (Query): One command of the query.
+            adj_to (int, optional): The command to refer to. Defaults to 0.
+
+        Returns:
+            Query: The completed, connected query.
+        """
+        spec.adj_to = self.adj_to + 1 if adj_to == 0 else adj_to
+        self.next = spec
+        return self
+
+    def command_properties(self, prop: str = "") -> List[str]:
+        """
+        Helper function to get the properties of all commands the query.
+
+        Args:
+            v (str, optional): _description_. Defaults to "".
+
+        Returns:
+            List[str]: Properties from commands in the order they should be executed.
+        """
+        chain = []
+        p = self
+        while p is not None:
+            chain.append(getattr(p, prop))
+            p = p.next
+        return chain
+
+    @classmethod
+    def spec(cls,
+             constraints: Constraints = None,
+             with_class: EntityType = EntityType.CUSTOM,
+             custom_class_name: str = "",
```

Comment:
do we have to differentiate "with_class" and "custom_class_name"? why not just use "with_class", and use EntityType.* when you want to use system defined classes (i.e. _Image, _Video, etc)?

Reply:
addressed this.